### DISCUSSION TITLE
Valheim-odin image is closed-source?

### DISCUSSION THREAD

**Discussion by mgan59** [mbround18/valheim-docker#278] [Q&A]
Stumbled onto this docker/valheim deployment repo and was reading through to understand how it all works.  I was confused though on how `odin` specifically factors in and how it works.  I noticed inside of the [Dockerfile:line5](https://github.com/mbround18/valheim-docker/blob/main/Dockerfile#L5) that it references:

```
FROM mbround18/valheim-odin:${ODIN_IMAGE_VERSION} as runtime
```

So that portion of the code is closed-source and proprietary?  I see the corresponding image on [dockerhub]  (https://hub.docker.com/r/mbround18/valheim-odin) with no link to a github source.

Just learning how this all works/snaps together.  Great project BTW really cool to see this.

**Comment by mbround18** [comment]
Nah it exists here in this same repo :) its just another image only containing odin for other purposes. For example, I dont use the built in cron jobs to check for updates and instead I use kubernetes cron jobs to update when ready. 

All source is under this repo and free to use within the scope of the [BSD 3-Clause "New" or "Revised" License](https://github.com/mbround18/valheim-docker/blob/main/LICENSE)  <3

**Reply by mbround18** [reply]
I updated the docs on the dockerhub for odin <3

**Comment by CosmicHorrorDev** [comment]
Yup as mentioned `odin` is also in this repo and is open source (I would hope so it's the main thing I've worked on 😅)

So `odin` is written in rust and his code lives in `src`. The dockerfile for him is `Dockerfile.odin` and you can see that the image also gets pushed before the valheim image in the workflows.
